### ISSUE TITLE
Remove usage of custom marshaller

### ISSUE THREAD

**Issue by alexandrudima** [Microsoft/vscode#40169]
*tl;dr* - You shall not use `URI` in signatures or the types referenced from `extHost.protocol.ts` anymore. Use `UriComponents` and `URI.revive` instead

We use a custom marshaller in the extension host <-> renderer communication layer. It treats `URI` and `RegExp` specially, and makes them like JSON primitive types. This is helpful, for example, when passing an argument of type `URI` over the wire, you know for a fact that on the receiving side, the object is an instance of URI.

However, **this is expensive**, as it needs to run on each and every object sent over the wire, in a deep, recursive manner. It is very easy to avoid using the custom marshaller, simply replace the usage of `URI` with `UriComponents` which is JSON friendly (an interface without methods). Then, in the implementation, you can use `URI.revive(...)` to get back a real `URI`.

For `RegExp` objects (extremely rare usage), you're gonna have to roll your own strategy.

All edits should result in removing `ProxyType.CustomMarshaller` from the proxies defined in `extHost.protocol.ts`.

See https://github.com/Microsoft/vscode/commit/b64355bf8da7379a30613c6e4f230c4c3679e798 for an example adoption

---
Remaining: 12 out of 49:
- [x] MainThreadCommands
- [x] MainThreadConfiguration @sandy081 
- [x] MainThreadDebugService @weinand 
- [x] MainThreadLanguageFeatures
- [x] MainThreadSCM @joaomoreno 
- [ ] MainThreadTask @dbaeumer 
- [x] ExtHostCommands
- [x] ExtHostConfiguration @sandy081 
- [x] ExtHostDebugService @weinand 
- [x] ExtHostLanguageFeatures
- [x] ExtHostSCM @joaomoreno 
- [ ] ExtHostTask @dbaeumer

**Comment by alexandrudima** [comment]
fyi @dbaeumer I adopted it for tasks via fabd8a57ec55d5de8e2c45c4be2752407c77f4e8

**Comment by alexandrudima** [comment]
Done and support removed.

**Comment by dbaeumer** [comment]
@alexandrudima thanks !
